src/clover/optimisation/appraisal.py

```diff
@@ -242,76 +243,82 @@ def _simulation_environmental_appraisal(  # pylint: disable=too-many-locals
+    #[5A,10A,100kWh,200kWh,300kWh,400kWh,1000kWh]
+    for GridTier.upper_bound_consumption in grid.tiers: #run over the list of tiers where we have the different upper bound consumption
     # Filter out based on whether the grid is current drawing or max-power in its pricing
-    if grids.type == GridType.CURRENT_DRAW:  # DIESEL GENERATOR
-        # Determine the peak current
-        if (
-            household_daily_peak_demand <= GridTier.upper_bound_consumption
-        ):  # A to kW and because peak so kWh so here I am saying if at anytime during a day the peak demand <=5A power then we only need tier 5A
-            tier_i_am_in = grid_type  # upper_bound-consumption_5
-        else:
-            tier_i_am_in = grid_type  # upper_bound-consumption_10
-    elif grids.type == GridType.DAILY_POWER:  # EDL
+        if grid.type == GridType.CURRENT_DRAW:  # DIESEL GENERATOR
+            if (
+                max(grid_energy)/voltage <= GridTier.upper_bound_consumption[0]
+            ):  # A to kW and because peak so kWh so here I am saying if at anytime during a day the peak demand <=5A power then we only need tier 5A
+                tier_i_am_in = grid.tiers[0]  # upper_bound-consumption_5
+            else:
+                tier_i_am_in = grid.tiers[1]  # upper_bound-consumption_10
+        elif grid.type == GridType.DAILY_POWER:  # EDL
         # Sum over the time period, you will need to code this somewhere, here would be fine for now
         # Determine the energy that was consumed
-        if household_monthly_demand <= upper_bound_tier_edl_1:
-            tier_i_am_in  # is the tier with ["upper_bound"]["consumption"]=100
-        elif (
-            household_monthly_demand
-            <= upper_bound_tier_edl_1 & household_monthly_demand
-            <= upper_bound_tier_edl_2
-        ):
-            tier_i_am_in  # is the tier with ["upper_bound"]["consumption"]=200
-        elif (
-            household_monthly_demand
-            <= upper_bound_tier_edl_2 & household_monthly_demand
-            <= upper_bound_tier_edl_3
-        ):
-            tier_i_am_in  # is the tier with ["upper_bound"]["consumption"]=300
-        elif (
-            household_monthly_demand
-            <= upper_bound_tier_edl_3 & household_monthly_demand
-            <= upper_bound_tier_edl_4
-        ):
-            tier_i_am_in  # is the tier with ["upper_bound"]["consumption"]=400
-        elif (
-            household_monthly_demand
-            <= upper_bound_tier_edl_4 & household_monthly_demand
-            <= upper_bound_tier_edl_5
-        ):
-            tier_i_am_in  # is the tier with ["upper_bound"]["consumption"]=1000
-    else:
-        raise Exception(
-            "Grid type must be one of {}".format(
-                ", ".join(
-                    {e.value for e in GridType}
-                )  # This will print all the allowed values
+            if household_monthly_demand <= GridTier.upper_bound_consumption[2]:
```

Comment:
## :bug::fire: Unneeded hardcoding
We came up with a solution that didn't need these hard-coded checks;
* We've got the tiers sorted based on their consumption,
* We know we're looking for the tier for which the total monthly demand is lower than the threshold (if they're upper bounds), and that we will be in the smallest tier for which this is true,
* We can use a for loop,
* which saves all these hard-coded if statements :smile:

Reply:
Okay will do that and let you know